File: src/main/java/org/oewntk/pojos/LemmaCS.java
/*
 * Copyright (c) 2021. Bernard Bou.
 */

package org.oewntk.pojos;

import java.util.Objects;

/**
 * Pair of lemma with cased (case-sensitive) form
 *
 * @author Bernard Bou
 */
public class LemmaCS
{
	public final Lemma lemma;

	public final NormalizedString cased;

	/**
	 * Constructor
	 *
	 * @param lemma lemma
	 * @param cased cased form
	 */
	public LemmaCS(final Lemma lemma, final NormalizedString cased)
	{
		this.lemma = lemma;
		this.cased = cased;
	}

	// I D E N T I T Y

	@Override
	public boolean equals(final Object o)
	{
		if (this == o)
		{
			return true;
		}
		if (o == null || getClass() != o.getClass())
		{
			return false;
		}
		LemmaCS that = (LemmaCS) o;
		return lemma.equals(that.lemma) && cased.equals(that.cased);
	}

	@Override
	public int hashCode()
	{
		return Objects.hash(lemma, cased);
	}

	@Override
	public String toString()
	{
		return this.cased.toString();
	}
}

File: src/main/java/org/oewntk/pojos/LemmaCS.kt
/*
 * Copyright (c) 2021-2024. Bernard Bou.
 */
package org.oewntk.pojos

import java.util.*

/**
 * Pair of lemma with cased (case-sensitive) form
 *
 * @property lemma lemma
 * @property cased cased form
 *
 * @author Bernard Bou
 */
class LemmaCS(
    val lemma: Lemma,
    private val cased: NormalizedString,
) {

    // I D E N T I T Y

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as LemmaCS
        return lemma == that.lemma && cased == that.cased
    }

    override fun hashCode(): Int {
        return Objects.hash(lemma, cased)
    }

    override fun toString(): String {
        return cased.toString()
    }
}
